### ISSUE TITLE
Some results trained with Adam and AdamW on Librispeech

### ISSUE THREAD

**Issue by luomingshuang** [k2-fsa/snowfall#228]
1. I use the script (mmi_att_transformer_train.py) as the basic experiment setting and use train-clean-100 as the training data. 
2. I use Adam or AdamW optimizer to replace Noam. 
3. I compare the performance from the three optimizers.
4. The results are based on different learning rate and decay weight.
5. All the results are based on single GPU.
<img width="332" alt="d85ac1ec8a3834f18a811ec4bce8d4e" src="https://user-images.githubusercontent.com/37799481/124563441-9efdc000-de72-11eb-8e58-9977f3e63d01.png">

<img width="552" alt="bb8879f70edf97ce244eb9fa7b3321b" src="https://user-images.githubusercontent.com/37799481/124563336-88576900-de72-11eb-9896-e4e7943329f2.png">

**Comment by danpovey** [comment]
OK, thanks.  I guess this confirms that we need the noam-type learning rate scheduler.
At some point I'd like to try something like Adam but with weight decay where it decays to the initial (random) value, not
to 0.

**Comment by luomingshuang** [comment]
Yes, maybe I can try to set the lr (learning rate) according to the epoch. When the epoch <=5 (or other value), setting the lr=0.0001. And when the epoch >5, setting the lr=0.00001.

**Comment by danpovey** [comment]
Study the Noam optimizer, how it sets the learning rate.
It is just a modification of Adam, to set a certain schedule.
